src/rust/engine/src/externs.rs

```diff
@@ -216,21 +219,41 @@ pub fn generator_send(generator: &Value, arg: &Value) -> Result<GeneratorRespons
     PyGeneratorResponseType::Throw => Err(PyResult::failure_from(response.values.unwrap_one())),
     PyGeneratorResponseType::Get => {
       let mut interns = INTERNS.write();
-      let constraint = TypeConstraint(interns.insert(response.constraints.unwrap_one()));
-      Ok(GeneratorResponse::Get(Get(
-        constraint,
-        interns.insert(response.values.unwrap_one()),
-      )))
+      let product = TypeConstraint(interns.insert(response.products.unwrap_one()));
+      let subject_declared_type =
```

Comment:
Hm. So I guess this is because at runtime (rather than at `@rule` parse time), we'll see a `Get` with a declared type that might be a union, and then we might want to know the members of the union in order to validate that the given value is a member of that union?

It feels like not actually checking the declared type here, and just using exactly the type of the subject value would be fine. The result would still be the same as if there had been multiple literal `Get`s in the body of the `@rule`... and by this point we would already have validated that those are correct in the rule graph.

AFAICT, that would mean this method would not need to change at all?

Reply:
If I'm parsing your comment correctly, after expanding the `Get`s to include all the registered constituent union members, this would not need to change.

Reply:
I have implemented this, all I had to do was make `gen_get()` return an error instead of panicking!